ext-apiserver/pkg/apis/cluster/v1alpha1/machineset_types.go

```diff
@@ -130,16 +131,39 @@ type MachineSetStatus struct {
 
 // Validate checks that an instance of MachineSet is well formed
 func (MachineSetStrategy) Validate(ctx request.Context, obj runtime.Object) field.ErrorList {
```

Comment:
usually this is placed close to a `registry` pkg, since we are using a framework to crate an api extension I am not sure whether this is a good place to put validation code ?  Also do I have to regenerate code ?